### PR TITLE
Upgrade Hudi 0.12.1 for the connector

### LINKED ISSUE #14430: Upgrade Hudi version for the connector
Current Hudi version for the trino-hudi connector is 0.11.1.
Hudi v0.12.1 contains a critical fix for [HUDI-4687](https://issues.apache.org/jira/browse/HUDI-4687).

### PR DESCRIPTION
**Pull request by albericgenius**
## Description
Fix #14430
Upgrade Hudi version for the connector 

## Release notes

( x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`14430`)
```

### CODE REVIEW

File: plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/testing/TpchHudiTablesInitializer.java [hunk 1]
```diff
@@ -200,7 +200,7 @@ private HoodieJavaWriteClient<HoodieAvroPayload> createWriteClient(TpchTable<?>
         }
 
         HoodieIndexConfig indexConfig = HoodieIndexConfig.newBuilder().withIndexType(HoodieIndex.IndexType.INMEMORY).build();
-        HoodieCompactionConfig compactionConfig = HoodieCompactionConfig.newBuilder().archiveCommitsWith(20, 30).build();
+        HoodieCompactionConfig compactionConfig = HoodieCompactionConfig.newBuilder().build();
```
Comment (codope) [inline plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/testing/TpchHudiTablesInitializer.java:203]: Can you build archival configs using `HoodieArchivalConfig` and set it to write config. I have a [commit](https://github.com/codope/trino/commit/d4e5d54f44afc316b0c2ccfb1d66aaaf08355298) upgrading hudi version to 0.12.1-rc1. You can take a look for reference.

File: plugin/trino-hudi/pom.xml [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
     <properties>
         <air.main.basedir>${project.parent.basedir}</air.main.basedir>
-        <dep.hudi.version>0.11.1</dep.hudi.version>
+        <dep.hudi.version>0.12.1</dep.hudi.version>
```
Comment (codope) [inline plugin/trino-hudi/pom.xml:18]: With this, I think we can also remove the `--add-opens` configs: https://github.com/trinodb/trino/blob/master/plugin/trino-hudi/pom.xml#L461-L467

File: plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/testing/TpchHudiTablesInitializer.java (hunk 1 above)
Comment (albericgenius) [reply plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/testing/TpchHudiTablesInitializer.java]: Thanks @codope , I am working with my colleage on MOR, there is reflect issue when we use 0.11.1, so i upgraded to 0.12.1 to resolve issue on JDK 17. Next time, I will talk with you first.

File: plugin/trino-hudi/pom.xml (hunk 2 above)
Comment (albericgenius) [reply plugin/trino-hudi/pom.xml:18]: Thanks for this information.

File: plugin/trino-hudi/pom.xml [hunk 3]
```diff
@@ -458,13 +462,6 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <configuration>
-                    <argLine>
-                        --add-opens=java.base/java.lang=ALL-UNNAMED
-                        --add-opens=java.base/java.util=ALL-UNNAMED
-                        --add-opens=java.base/java.util.concurrent=ALL-UNNAMED
```
Comment (findepi) [inline plugin/trino-hudi/pom.xml:465]: thank you

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
@codope any other changes?

**Review by codope — APPROVED** [review]
Looks good to me.

**Pull request merged**